components/user_settings/manage_languages.jsx

```diff
@@ -43,8 +43,10 @@ export default class ManageLanguage extends React.Component {
 
         updateUser(user, Constants.UserUpdateEvents.LANGUAGE,
             () => {
-                GlobalActions.newLocalizationSelected(user.locale);
                 this.setState({isSaving: false});
+                setTimeout(() => {
+                    GlobalActions.newLocalizationSelected(user.locale);
+                }, 0);
```

Comment:
Can you please explain what's going on here?

Reply:
Based on my observation the JS warning occurred when the `GlobalActions` (root unmount then mount) is called before `setState`. 
Rearranging the two seemed to fix the warning but not fully since `setState` is "async maybe". As a result, I'm still seeing intermittent warning. 
That's why I thought wrapping `GlobalActions` with `setTimeout of 0` will guarantee to make it at the end of execution queue.
I'm 0/5 here and I might be wrong.

Reply:
Could we remove setState call here? Anyway, its part of react state and not redux which gets destroyed on closing the modal.  Why not just close the modal with `GlobalActions.newLocalizationSelected` and let it be.

Reply:
hmmm, okay I guess it makes sense, PR updated. Thanks!